src/security/Certificate.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 1996-2022 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_SECURITY_CERTIFICATE_H
+#define SQUID_SRC_SECURITY_CERTIFICATE_H
+
+#include "security/forward.h"
+
+// The accessing/testing functions below require a non-constant Certificate when
+// it is modified by an underlying library implementation (e.g., GnuTLS).
+
+namespace Security
+{
+
+/// The SubjectName field of the given certificate (if found) or an empty SBuf.
+SBuf SubjectName(Certificate &);
+
+/// The Issuer field of the given certificate (if found) or an empty SBuf.
+SBuf IssuerName(Certificate &);
+
+/// \returns whether cert was (correctly) issued by the given issuer
+/// Due to complexity of the underlying checks, it is impossible to clearly
+/// distinguish pure negative answers (e.g., two independent certificates)
+/// from errors (e.g., the issuer certificate lacks the right CA extension).
+bool IsIssuedBy(Certificate &cert, Certificate &issuer);
```

Comment:
Please let me know if you prefer other spellings for this function name. Here are a few reasonable variants:

1. IssuedBy(cert, issuer)
2. IssuerOf(issuer, cert) -- matches OpenSSL X509_check_issued() argument order
3. Issued(issuer, cert) -- matches OpenSSL X509_check_issued() argument order
4. AissuedB(issuer, cert) -- matches OpenSSL X509_check_issued() argument order; violates CamelCase for readability sake
5. AissuedByB(cert, issuer) -- violates CamelCase for readability sake

I am actually leaning towards IssuedBy() (and SelfSigned() below). IssuerOf() also looks good. There is no reason to use an "Is" prefix in this context -- the spelling cannot be mistaken for a command or request to modify the certificate; it is clearly a test or check.

I am not a fan of the last two, but will play along if others insist on one of them.

Reply:
> I am actually leaning towards IssuedBy() (and SelfSigned() below). 

likewise.

Reply:
Renamed in 0dbbf52. I also polished their descriptions a little in 9b02b6b.